ap_git/core.py

```diff
@@ -648,3 +650,93 @@ def shallow_clone_at_commit_from_local(source: str,
         source_repo.__branch_delete(branch_name=temp_branch_name, force=True)
 
         return cloned_repo
+
+    @staticmethod
+    def from_path(path: str,
+                  clone_from: str = None,
+                  recurse_submodules: bool = False,
+                  shallow_submodules: bool = False,
+                  clone_if_broken: bool = False,
+                  clone_if_not_found: bool = False) -> "GitRepo":
+        """
+        Initialise a GitRepo with a given path to a local repository.
+        Clone from the given source if a git repository does not exist
+        at the given path or it is broken (i.e., something exists at the
+        path but is not a git repository).
+        Note that the files at the path are cleaned up before cloning in
+        second case.
+
+        Parameters:
+            path (str): Source path of the local git repository.
+            clone_from (str): Url/Path to remote repository to clone from.
+                              (defaults to None)
+            recurse_submodules (bool): Recurse into submodules (default is
+                                       False)
+            shallow_submodules (bool): any cloned submodules will be shallow.
+                                       (default is False)
+            clone_if_broken (bool): Clone from source if local repository looks
+                                    broken. (default is False)
+            clone_if_not_found (bool): Clone from source if local repository is
+                                       not found. (default is False)
+
+        Returns:
+            GitRepo: the GitRepo object for the local git repository.
+
+        Raises:
+            ValueError: If clone_from is not provided and a clone attempt is
+                        made.
+            FileNotFoundError: If local git repository is not found and
+                               clone_if_not_found is disabled.
+            NonGitDirectoryError: If the directory at the given path is not
+                                  a git repository and clone_if_broken is
+                                  disabled.
```

Comment:
I don't know if this is a good option.  The user should be able to fix it themselves - it should be a one-off error any time it occurs.

The rmtree involved in this makes me nervious - easy way for someone to lose their work if they stuff up a single command-line...

Reply:
My thought process was that the ArduPilot repository in the base directory wouldn't contain anything important to the user that couldn't be brought back if deleted. I also included a warning in the method description. But yes, it can be disastrous if we plan it to use at other places in future. I think we should leave it for the user to fix.